main.go

```diff
@@ -257,6 +251,52 @@ func main() {
 		klog.Fatalf("error syncing nftables rules: %v", err)
 	}
 
+	mux := http.NewServeMux()
+	mux.HandleFunc("/healthz", func(w http.ResponseWriter, r *http.Request) {
+		err := checkHealth()
+		if err != nil {
+			http.Error(w, err.Error(), http.StatusInternalServerError)
+			return
+		}
+
+		_, err = w.Write([]byte("ok"))
+		if err != nil {
+			klog.Fatalf("error writing HTTP response on health endpoint on successful health check")
+		}
+	})
+	mux.Handle("/metrics", promhttp.Handler())
+
+	// run daemon server
+	go func() {
+		klog.Infof("starting daemon server listening in %s", serverBindAddress)
+		http.ListenAndServe(serverBindAddress, mux) // nolint:errcheck
+	}()
+
+	go func() {
+		ticker := time.NewTicker(syncInterval)
+		defer ticker.Stop()
+		klog.Infof("Start auto sync nftables rules")
+
+		for {
+			if ctx.Err() != nil {
+				klog.Infof("Stopping auto sync")
+				return
+			}
+
+			if err := syncRules(v4net, v6net, gwIface); err != nil {
+				klog.Fatalf("auto sync, syncRules error: %v", err)
```

Comment:
this was my bad I think , that I give that example, but since we are retrying it seems safer to just log and retry

```suggestion
				klog.Infof("auto sync, syncRules error: %v", err)
```